### ISSUE TITLE
[Accessibility] Tinkering with `disabled` and `readonly` input fields 

### ISSUE THREAD

**Issue by gpeuc** [SAP/open-ux-tools#801]
## Intro

During one of our thinking sessions, we came to a realisation that we have a big issue with `disabled` input fields at the moment. The nature of HTML prevents those fields to be focused via keyboard at all. They are just skipped. This in turn means that people with sight issues won't even be aware that on a certain page there could be a number of these `disabled` fields, as they cannot ever focus them with a keyboard. This is not our bug as such, this is simply how HTML `disabled` works.

However, there is a suggested workaround that seems to be a common recommendation all over the accessibility design world - swap `disabled` into `readonly`.

`readonly` fields cannot be modified, however, they are keyboard accessible.

## Versions

We need to update & build some new flavours of our input fields.

**1. Regular input field**

That's the one we have at the moment, it works great. It's fully enabled, keyboard accessible, responds to mouse-hover, etc.
![image](https://user-images.githubusercontent.com/52239744/202478675-302fa661-338f-4327-ac74-05f51127bac5.png)
Parameters
- border: vscode-editorWidget-border
- text: vscode-input-foreground
- placeholder text: vscode-input-placeholderForeground
- background: vscode-input-background

Hover:
![image](https://user-images.githubusercontent.com/52239744/202488118-cfdd9ecc-2915-459b-9615-89c0ae36eb29.png)
Parameter: border - vscode-focusBorder

**2. Disabled input field**

Due to some possibility we actually need, for some reason, an actual disabled input field, we can keep them in our toolkit. Let it just be there, why not. Disabled input fields have disabled label as well.
![image](https://user-images.githubusercontent.com/52239744/202478777-155aa080-5f71-4f4a-b552-5c9dd4d4c6e5.png)
Parameters;
Same as regular state (Placeholder variant)
Opacity: 50%
Hover: None

**3. ⭐ NEW → Read only field with content**

In some places we do require that users can see and select the content of an input field, but not modify it. Label is enabled.

The styling: 
![image](https://user-images.githubusercontent.com/52239744/202478589-3597f05b-0573-4789-85e0-74ea1ce13733.png)
Parameters: 

- border: vscode-editorWidget-border
 -  border style dashed, (dash 2, gap 2)
- text: vscode-input-foreground
 - text style, italic 
- background: vscode-editor-background

Hover: Parameter: border - vscode-focusBorder
![image](https://user-images.githubusercontent.com/52239744/202488245-d3035f9c-412a-4264-8c5b-29d1450ccae0.png)
Focus: border style: solid.
![image](https://user-images.githubusercontent.com/52239744/202488327-436f78a3-d19f-42bf-88f7-a77140ec41d1.png)

**4. ⭐ NEW → Read only field _without_ content**

This is what **most of our currently disabled input fields should become**. There should be a placeholder text, and since it's `readonly`, users can focus into it with a keyboard. They can't do anything here, can't even select the placeholder text (because that's how placeholders work), but this allows us to provide screen readers with some good stuff to read out when the users focus in. At least users can dip into it and know what's happening. Label is disabled.

The styling:
![image](https://user-images.githubusercontent.com/52239744/202479035-15578bc3-e34a-4c58-a1a1-f05caa722966.png)

Parameters: Visual same as existing disabled state
Hover: None
Focus: vscode-focusBorder
![image](https://user-images.githubusercontent.com/52239744/202488734-89f464c0-a127-43b0-b2f2-b7345acb43ba.png)

## All of them together
![image](https://user-images.githubusercontent.com/52239744/202487956-04a351fc-ce58-4d39-b9bf-807e623f9dfb.png)
[Design Spec](https://www.figma.com/file/tjmZpoyHzi61Rm1fm1gJ5u/Fiori-tools-UI-Kit?node-id=4002%3A11916&t=QIKm2MOh1KiRYmJa-1)

**Comment by gpeuc** [comment]
There is then an additional thing: aria descriptions of these fields. Now that users can keyboard focus into them we have to read something out for screen readers. We should most likely start with “Disabled field”  so that blind users know immediately that this thing is inactive. And on top of that, we should put in description the contents of the info bubble, because you cannot reach that thing with keyboard as well.

So for Navigation Path, aria-description could be something like → 

“Disabled field Navigation Path. Values are dependent on Entity Type”

<img width="600" alt="Screenshot 2022-11-24 at 15 16 02" src="https://user-images.githubusercontent.com/44162846/203817761-89e8faf8-36a5-4c29-b6b3-9adf3ea687ab.png">

**Comment by 815are** [comment]
PR with changes for `UITextInput`, `UIComboBox`, `UIDropdown`, `UITreeDropdown` - https://github.com/SAP/open-ux-tools/pull/859
